ocaml/xapi/xapi_pif.ml

```diff
@@ -840,11 +857,23 @@ let rec plug ~__context ~self =
       let phy_pif = Db.Network_sriov.get_physical_PIF ~__context ~self:sriov in
       debug "PIF is SRIOV logical PIF, also bringing up SRIOV physical PIF";
       plug ~__context ~self:phy_pif
-    | Physical pif_rec when pif_rec.API.pIF_bond_slave_of <> Ref.null ->
-      raise Api_errors.(Server_error (cannot_plug_bond_slave, [Ref.string_of self]))
+    | Physical pif_rec ->
+      let bond = pif_rec.API.pIF_bond_slave_of in
+      if bond <> Ref.null then begin
+        if pif_rec.API.pIF_sriov_physical_PIF_of <> [] then begin
+          (* It's a bond slave and SR-IOV physical *)
+          let bond_master_pif = Db.Bond.get_master ~__context ~self:bond in
+          debug "PIF is SRIOV physical PIF and bond slave, also bringing up bond master PIF";
+          plug ~__context ~self:bond_master_pif
```

Comment:
Here we are bringing up the master, does this automatically bring up the slave as well? Or we still need a plug for the slave here?

Reply:
Yes. It will bring up slaves automatically.